com.unity.render-pipelines.universal/Editor/Camera/UniversalRenderPipelineCameraStackUI.cs

```diff
@@ -0,0 +1,296 @@
+            // Add here the supported types of cameras
+        };
+
+        static class Styles
+        {
+            public const int cameraNameWidth = 150;
+            public const int errorIconWidth = 12;
+
+            public static readonly Vector2 errorIconSize = new Vector2(errorIconWidth, errorIconWidth);
+            public static readonly GUIStyle errorStyle = new GUIStyle(EditorStyles.label) { padding = new RectOffset { left = -errorIconWidth }};
+
+            static readonly Texture2D errorIcon = EditorGUIUtility.Load("icons/console.erroricon.sml.png") as Texture2D;
+            public static readonly Dictionary<CameraRenderType, GUIContent> typeNotSupportedMessage = Enum.GetValues(typeof(CameraRenderType))
+                .Cast<CameraRenderType>()
+                .Where(cameraRenderType => !s_ValidCameraTypes.Contains(cameraRenderType))
+                .ToDictionary(cameraRenderType => cameraRenderType, cameraRenderType => EditorGUIUtility.TrTextContent(string.Empty, cameraRenderType.GetName() + " is not supported", errorIcon));
+
+            public static readonly GUIContent[] camerasToAddNotFound = { EditorGUIUtility.TrTextContent("Could not find suitable cameras") };
+            public static readonly GUIContent PP = EditorGUIUtility.TrTextContent("PP", "The camera has enabled the Post Processing");
+
+            public static readonly string validTypes = $"Valid types are: {string.Join(",", s_ValidCameraTypes.Select(i => $"{i}"))}";
+        }
+
+        readonly Camera m_Camera;
+        readonly ReorderableList m_LayerList;
+        readonly UniversalRenderPipelineSerializedCamera m_SerializedCamera;
+
+        bool CanRemove(ReorderableList list)
+        {
+            // As the list can delete the last item, allow the delete action if there is at least one item
+            return m_SerializedCamera.cameras.arraySize > 0;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="Camera"/> at a given index
+        /// </summary>
+        /// <param name="index">The index to fetch the camera</param>
+        /// <returns><see cref="Camera"/> if the index is valid, otherwise returns null</returns>
+        public Camera this[int index]
+        {
+            get
+            {
+                if (index < 0 || index > m_SerializedCamera.cameras.arraySize - 1)
+                    return null;
+
+                // Return the camera on that index
+                return m_SerializedCamera.cameras
+                    .GetArrayElementAtIndex(index)
+                    .objectReferenceValue as Camera;
+            }
+        }
+
+        void RemoveCameraByIndexFromList(int selectedIndex)
+        {
+            Camera cam = this[selectedIndex];
+            if (cam == null)
+                return;
+
+            // Delete the camera from the object
+            m_SerializedCamera.cameras.DeleteArrayElementAtIndex(selectedIndex);
+        }
+
+        void RemoveCameraFromList(ReorderableList list)
```

Comment:
should we rename to RemoveSelectedCamerasFromList ?(this way it is obvious you dont need to pass an index)